### PR TITLE
Add prod cd workflow

### PR DESCRIPTION
**Pull request by SpencerLN**

### CODE REVIEW

File: .github/workflows/build-and-publish-prod.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          env-file: .env
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build Maven with Docker Profile
+        run: mvn clean install -Pdocker -DskipTests -B
+      - name: Build and push
+        uses: docker/bake-action@master
```
Comment (ismisepaul) [inline .github/workflows/build-and-publish-prod.yml:48]: `uses: docker/bake-action@e878ca2dcabf3e92e501fbbe72cda83e0046e0de  # v2.2.0`

**Review by ismisepaul — COMMENTED** [review]
LGTM my only suggestion is we pin the actions following the [hardening guide](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) from Github

What do you think?

**Comment by SpencerLN** [comment]
> LGTM my only suggestion is we pin the actions following the [hardening guide](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) from Github
> 
> What do you think?

I think that for the actions published by GitHub, Docker, Google, or other well-known publishers it is pretty safe to pin to a specific branch/tag instead of a specific commit. This allows us to automatically receive bug/security fixes for these actions.

For actions published by less established publishers, I think pinning to the specific commit makes sense. In this case, I think the `c-py/action-dotenv-to-setenv` action is the only one that falls into this category.

**Review by ismisepaul — APPROVED** [review]
Agreed. 

LGTM

**Pull request merged**